packages/schematics/src/index.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {FilePredicate, MergeStrategy} from './tree/interface';
+import {Tree as TreeInterface} from './tree/interface';
+import {branch, empty, merge, optimize, partition} from './tree/static';
+
+
+export { BaseException } from './exception/exception';
+
+export * from './tree/action';
+export * from './engine/collection';
+export * from './engine/engine';
+export * from './engine/interface';
+export * from './tree/interface';
+export * from './rules/base';
+export * from './rules/move';
+export * from './rules/random';
+export * from './rules/schematic';
+export * from './rules/template';
+export * from './rules/url';
+export * from './tree/empty';
+export * from './tree/filesystem';
+export * from './tree/memory-host';
+export {UpdateRecorder} from './tree/interface';
+export * from './engine/schematic';
+export * from './sink/dryrun';
+export {FileSystemSink} from './sink/filesystem';
+export * from './utility/path';
+
+
+export interface TreeConstructor {
```

Comment:
Needs to be done where it's exported, but not in the interface definition (there's a circular dependency).